cookbook/security/api_key_authentication.rst

```diff
@@ -67,7 +60,7 @@ value and then a User object is created::
         public function authenticateToken(TokenInterface $token, UserProviderInterface $userProvider, $providerKey)
         {
             $apiKey = $token->getCredentials();
-            $username = $this->userProvider->getUsernameForApiKey($apiKey);
+            $username = $userProvider->getUsernameForApiKey($apiKey);
```

Comment:
@xabbuh  I like your first question, but I think it is out of scope of this topic. Even without my changes, same question could be asked. The cookbook referenced in this PR says `If supportsToken() returns true, Symfony will now call authenticateToken()` (in explanation how authentication process works). So Symfony checks that automatically. Would be strange if it wouldn't check. IMHO when method is defined in interface it's so not because you _should_ use it internally.

Thanks for the note about user provider, I agree that we should add check here. I will update my PR later.